isobus/src/isobus_task_controller_server.cpp

```diff
@@ -0,0 +1,1057 @@
+	}
+
+	bool TaskControllerServer::send_measurement_command(std::shared_ptr<ControlFunction> clientControlFunction, std::uint8_t commandValue, std::uint16_t dataDescriptionIndex, std::uint16_t elementNumber, std::uint32_t processDataValue) const
+	{
+		bool retVal = false;
+
+		if (nullptr != clientControlFunction)
+		{
+			std::array<std::uint8_t, CAN_DATA_LENGTH> payload = { 0 };
+			payload[0] = (commandValue & 0x0F);
+			payload[0] |= (elementNumber && 0x0F) << 4;
+			payload[1] = static_cast<std::uint8_t>((elementNumber >> 4) & 0xFF);
+			payload[2] = static_cast<std::uint8_t>(dataDescriptionIndex & 0xFF);
+			payload[3] = static_cast<std::uint8_t>((dataDescriptionIndex >> 8) & 0xFF);
+			payload[4] = static_cast<std::uint8_t>(processDataValue & 0xFF);
+			payload[5] = static_cast<std::uint8_t>((processDataValue >> 8) & 0xFF);
+			payload[6] = static_cast<std::uint8_t>((processDataValue >> 16) & 0xFF);
+			payload[7] = static_cast<std::uint8_t>((processDataValue >> 24) & 0xFF);
+
+			retVal = CANNetworkManager::CANNetwork.send_can_message(static_cast<std::uint32_t>(CANLibParameterGroupNumber::ProcessData),
+			                                                        payload.data(),
+			                                                        payload.size(),
+			                                                        serverControlFunction,
+			                                                        clientControlFunction,
+			                                                        static_cast<std::uint8_t>(ProcessDataCommands::SetValueAndAcknowledge) == commandValue ? CANIdentifier::CANPriority::Priority3 : CANIdentifier::CANPriority::Priority5);
+		}
+		return retVal;
+	}
+
+	bool TaskControllerServer::send_status_message() const
+	{
+		std::array<std::uint8_t, CAN_DATA_LENGTH> payload = {
+			static_cast<std::uint8_t>(ProcessDataCommands::Status) | 0xF0U,
+			0xFF,
+			0xFF,
+			0xFF,
+			currentStatusByte,
+			currentCommandSourceAddress,
+			currentCommandByte,
+			0xFF
+		};
+
+		return CANNetworkManager::CANNetwork.send_can_message(static_cast<std::uint32_t>(CANLibParameterGroupNumber::ProcessData),
+		                                                      payload.data(),
+		                                                      payload.size(),
+		                                                      serverControlFunction,
+		                                                      nullptr,
+		                                                      CANIdentifier::CANPriority::Priority3);
+	}
+
+	bool TaskControllerServer::send_version(std::shared_ptr<ControlFunction> clientControlFunction) const
+	{
+		std::array<std::uint8_t, CAN_DATA_LENGTH> payload = {
+			static_cast<std::uint8_t>(TechnicalDataCommandParameters::ParameterVersion) << 4,
+			static_cast<std::uint8_t>(TaskControllerVersion::SecondPublishedEdition),
```

Comment:
issue: the version needs to be (optionally) configurable. If a TC were to support only the "TC-SC generation 1" it needs to send the version number 3, instead of the 4 here.

Reply:
Added as an optional parameter to the constructor of the server